### PR TITLE
Branch a

### PR DESCRIPTION
**Pull request by Meri-MG**
#### In this PR was:
- Written unit tests using Rspec.
- Added tests to the codebase.
- Written Rspec unit tests with mocks.
- Used the TDD concept.

### CODE REVIEW

File: solver.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Solver
+  def reverse(str)
+    str = str.reverse
+  end
+end
```
Comment (mmsesay) [inline solver.rb]: - [ ] By the look of things you wrote the method before writing the tests. As the name implies TDD, you need to write the tests first before the method implementation. 

See screenshot of your commit history:
a1

File: solver.rb [hunk 2]
```diff
@@ -2,4 +2,10 @@ class Solver
   def reverse(str)
     str = str.reverse
   end
+
+  def fizzbuz(num)
+    if num % 3 == 0 && num % 5 == 0
+      'fizzbuzz'
+    end
```
Comment (mmsesay) [inline solver.rb]: - [ ] Found this again, you created the method before writing the tests. You need to write the tests first, ensure they pass before writing the method in the solver class.

Method commit screenshot:
a2

Test commit screenshot.
a3

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @mwanawabangona & @Meri-MG, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Dipeshtwis — APPROVED** [review]
Hi @Meri-MG & @mwanawabangona, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**